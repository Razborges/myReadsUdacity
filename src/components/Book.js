import React from 'react'
import PropTypes from 'prop-types'

const Book = (props) => (
  <li key={props.book.id}>
    <div className="book">
      <div className="book-top">
        <div className="book-cover" style={{ width: 128, height: 193, backgroundImage: `url(${props.book.imageLinks.smallThumbnail})` }} />
        <div className="book-shelf-changer">
          <select value={props.book.shelf} onChange={(e) => props.changeBook(props.book, e.target.value)}>
            <option value="none">Move to...</option>
            <option value="currentlyReading">Currently Reading</option>
            <option value="wantToRead">Want to Read</option>
            <option value="read">Read</option>
            <option value="none">None</option>
          </select>
        </div>
      </div>
      <div className="book-title">{props.book.title}</div>
      { props.book.authors && props.book.authors.map((author, index) => (
          <div className="book-authors" key={index}>{author}</div>
      ))}
    </div>
  </li>
)

Book.propTypes = {
  book: PropTypes.object.isRequired,
  changeBook: PropTypes.func.isRequired
}

export default Book