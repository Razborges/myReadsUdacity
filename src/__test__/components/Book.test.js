import React from 'react';
import { shallow, mount } from 'enzyme';
import Book from './../../components/Book';

import booksMock from './../__mock__/booksMock';


describe('<Book />', () => {
  const changeBook = jest.fn();
  const wrapper = mount(<Book book={booksMock.books[0]} changeBook={changeBook} />);

  it('shallow renders without crashing', () => {
    expect(shallow(<Book book={booksMock.books[0]} changeBook={jest.fn()} />))
  });

  it('snapshot Book', () => {
    expect(shallow(<Book book={booksMock.books[0]} changeBook={jest.fn()} />)).toMatchSnapshot()
  });

  it('verify options', () => {
    expect(wrapper.find('option').length).toBe(5);
  });

  it('verify selected option', () => {
    expect(wrapper.find('select').props().value).toBe('read');
  });

  it('modify option', () => {
    wrapper.find('select').simulate('change', { target: { value: 'currentlyReading' }});
    expect(changeBook).toHaveBeenCalledTimes(1)
  });
})
