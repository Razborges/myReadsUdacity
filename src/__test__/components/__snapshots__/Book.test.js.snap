exports[`<Book /> snapshot Book 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Book
    book={
        Object {
            "allowAnonLogging": true,
            "authors": Array [
              "William E. Shotts, Jr.",
            ],
            "averageRating": 4,
            "canonicalVolumeLink": "https://market.android.com/details?id=book-nggnmAEACAAJ",
            "categories": Array [
              "COMPUTERS",
            ],
            "contentVersion": "1.2.2.0.preview.2",
            "description": "You\'ve experienced the shiny, point-and-click surface of your Linux computer—now dive below and explore its depths with the power of the command line. The Linux Command Line takes you from your very first terminal keystrokes to writing full programs in Bash, the most popular Linux shell. Along the way you\'ll learn the timeless skills handed down by generations of gray-bearded, mouse-shunning gurus: file navigation, environment configuration, command chaining, pattern matching with regular expressions, and more. In addition to that practical knowledge, author William Shotts reveals the philosophy behind these tools and the rich heritage that your desktop Linux machine has inherited from Unix supercomputers of yore. As you make your way through the book\'s short, easily-digestible chapters, you\'ll learn how to: * Create and delete files, directories, and symlinks * Administer your system, including networking, package installation, and process management * Use standard input and output, redirection, and pipelines * Edit files with Vi, the world’s most popular text editor * Write shell scripts to automate common or boring tasks * Slice and dice text files with cut, paste, grep, patch, and sed Once you overcome your initial \"shell shock,\" you\'ll find that the command line is a natural and expressive way to communicate with your computer. Just don\'t be surprised if your mouse starts to gather dust. A featured resource in the Linux Foundation\'s \"Evolution of a SysAdmin\"",
            "id": "nggnmAEACAAJ",
            "imageLinks": Object {
              "smallThumbnail": "http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api",
              "thumbnail": "http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api",
            },
            "industryIdentifiers": Array [
              Object {
                "identifier": "9781593273897",
                "type": "ISBN_13",
              },
              Object {
                "identifier": "1593273894",
                "type": "ISBN_10",
              },
            ],
            "infoLink": "https://play.google.com/store/books/details?id=nggnmAEACAAJ&source=gbs_api",
            "language": "en",
            "maturityRating": "NOT_MATURE",
            "pageCount": 480,
            "panelizationSummary": Object {
              "containsEpubBubbles": false,
              "containsImageBubbles": false,
            },
            "previewLink": "http://books.google.com/books?id=nggnmAEACAAJ&dq=linux&hl=&cd=3&source=gbs_api",
            "printType": "BOOK",
            "publishedDate": "2012",
            "publisher": "No Starch Press",
            "ratingsCount": 2,
            "readingModes": Object {
              "image": false,
              "text": true,
            },
            "shelf": "read",
            "subtitle": "A Complete Introduction",
            "title": "The Linux Command Line",
          }
    }
    changeBook={[Function]} />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "nggnmAEACAAJ",
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="book">
        <div
                className="book-top">
                <div
                        className="book-cover"
                        style={
                                Object {
                                        "backgroundImage": "url(http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api)",
                                        "height": 193,
                                        "width": 128,
                                      }
                        } />
                <div
                        className="book-shelf-changer">
                        <select
                                onChange={[Function]}
                                value="read">
                                <option
                                        value="none">
                                        Move to...
                                </option>
                                <option
                                        value="currentlyReading">
                                        Currently Reading
                                </option>
                                <option
                                        value="wantToRead">
                                        Want to Read
                                </option>
                                <option
                                        value="read">
                                        Read
                                </option>
                                <option
                                        value="none">
                                        None
                                </option>
                        </select>
                </div>
        </div>
        <div
                className="book-title">
                The Linux Command Line
        </div>
        <div
                className="book-authors">
                William E. Shotts, Jr.
        </div>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="book-top">
            <div
                        className="book-cover"
                        style={
                                    Object {
                                                "backgroundImage": "url(http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api)",
                                                "height": 193,
                                                "width": 128,
                                              }
                        } />
            <div
                        className="book-shelf-changer">
                        <select
                                    onChange={[Function]}
                                    value="read">
                                    <option
                                                value="none">
                                                Move to...
                                    </option>
                                    <option
                                                value="currentlyReading">
                                                Currently Reading
                                    </option>
                                    <option
                                                value="wantToRead">
                                                Want to Read
                                    </option>
                                    <option
                                                value="read">
                                                Read
                                    </option>
                                    <option
                                                value="none">
                                                None
                                    </option>
                        </select>
            </div>
</div>,
          <div
            className="book-title">
            The Linux Command Line
</div>,
          Array [
            <div
              className="book-authors">
              William E. Shotts, Jr.
</div>,
          ],
        ],
        "className": "book",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="book-cover"
                style={
                                Object {
                                                "backgroundImage": "url(http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api)",
                                                "height": 193,
                                                "width": 128,
                                              }
                } />,
              <div
                className="book-shelf-changer">
                <select
                                onChange={[Function]}
                                value="read">
                                <option
                                                value="none">
                                                Move to...
                                </option>
                                <option
                                                value="currentlyReading">
                                                Currently Reading
                                </option>
                                <option
                                                value="wantToRead">
                                                Want to Read
                                </option>
                                <option
                                                value="read">
                                                Read
                                </option>
                                <option
                                                value="none">
                                                None
                                </option>
                </select>
</div>,
            ],
            "className": "book-top",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "book-cover",
                "style": Object {
                  "backgroundImage": "url(http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api)",
                  "height": 193,
                  "width": 128,
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <select
                  onChange={[Function]}
                  value="read">
                  <option
                                    value="none">
                                    Move to...
                  </option>
                  <option
                                    value="currentlyReading">
                                    Currently Reading
                  </option>
                  <option
                                    value="wantToRead">
                                    Want to Read
                  </option>
                  <option
                                    value="read">
                                    Read
                  </option>
                  <option
                                    value="none">
                                    None
                  </option>
</select>,
                "className": "book-shelf-changer",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <option
                      value="none">
                      Move to...
</option>,
                    <option
                      value="currentlyReading">
                      Currently Reading
</option>,
                    <option
                      value="wantToRead">
                      Want to Read
</option>,
                    <option
                      value="read">
                      Read
</option>,
                    <option
                      value="none">
                      None
</option>,
                  ],
                  "onChange": [Function],
                  "value": "read",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Move to...",
                      "value": "none",
                    },
                    "ref": null,
                    "rendered": "Move to...",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Currently Reading",
                      "value": "currentlyReading",
                    },
                    "ref": null,
                    "rendered": "Currently Reading",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Want to Read",
                      "value": "wantToRead",
                    },
                    "ref": null,
                    "rendered": "Want to Read",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Read",
                      "value": "read",
                    },
                    "ref": null,
                    "rendered": "Read",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "None",
                      "value": "none",
                    },
                    "ref": null,
                    "rendered": "None",
                    "type": "option",
                  },
                ],
                "type": "select",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "The Linux Command Line",
            "className": "book-title",
          },
          "ref": null,
          "rendered": "The Linux Command Line",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": "William E. Shotts, Jr.",
            "className": "book-authors",
          },
          "ref": null,
          "rendered": "William E. Shotts, Jr.",
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "nggnmAEACAAJ",
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="book">
          <div
                    className="book-top">
                    <div
                              className="book-cover"
                              style={
                                        Object {
                                                  "backgroundImage": "url(http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api)",
                                                  "height": 193,
                                                  "width": 128,
                                                }
                              } />
                    <div
                              className="book-shelf-changer">
                              <select
                                        onChange={[Function]}
                                        value="read">
                                        <option
                                                  value="none">
                                                  Move to...
                                        </option>
                                        <option
                                                  value="currentlyReading">
                                                  Currently Reading
                                        </option>
                                        <option
                                                  value="wantToRead">
                                                  Want to Read
                                        </option>
                                        <option
                                                  value="read">
                                                  Read
                                        </option>
                                        <option
                                                  value="none">
                                                  None
                                        </option>
                              </select>
                    </div>
          </div>
          <div
                    className="book-title">
                    The Linux Command Line
          </div>
          <div
                    className="book-authors">
                    William E. Shotts, Jr.
          </div>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="book-top">
              <div
                            className="book-cover"
                            style={
                                          Object {
                                                        "backgroundImage": "url(http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api)",
                                                        "height": 193,
                                                        "width": 128,
                                                      }
                            } />
              <div
                            className="book-shelf-changer">
                            <select
                                          onChange={[Function]}
                                          value="read">
                                          <option
                                                        value="none">
                                                        Move to...
                                          </option>
                                          <option
                                                        value="currentlyReading">
                                                        Currently Reading
                                          </option>
                                          <option
                                                        value="wantToRead">
                                                        Want to Read
                                          </option>
                                          <option
                                                        value="read">
                                                        Read
                                          </option>
                                          <option
                                                        value="none">
                                                        None
                                          </option>
                            </select>
              </div>
</div>,
            <div
              className="book-title">
              The Linux Command Line
</div>,
            Array [
              <div
                className="book-authors">
                William E. Shotts, Jr.
</div>,
            ],
          ],
          "className": "book",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="book-cover"
                  style={
                                    Object {
                                                      "backgroundImage": "url(http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api)",
                                                      "height": 193,
                                                      "width": 128,
                                                    }
                  } />,
                <div
                  className="book-shelf-changer">
                  <select
                                    onChange={[Function]}
                                    value="read">
                                    <option
                                                      value="none">
                                                      Move to...
                                    </option>
                                    <option
                                                      value="currentlyReading">
                                                      Currently Reading
                                    </option>
                                    <option
                                                      value="wantToRead">
                                                      Want to Read
                                    </option>
                                    <option
                                                      value="read">
                                                      Read
                                    </option>
                                    <option
                                                      value="none">
                                                      None
                                    </option>
                  </select>
</div>,
              ],
              "className": "book-top",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "book-cover",
                  "style": Object {
                    "backgroundImage": "url(http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api)",
                    "height": 193,
                    "width": 128,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <select
                    onChange={[Function]}
                    value="read">
                    <option
                                        value="none">
                                        Move to...
                    </option>
                    <option
                                        value="currentlyReading">
                                        Currently Reading
                    </option>
                    <option
                                        value="wantToRead">
                                        Want to Read
                    </option>
                    <option
                                        value="read">
                                        Read
                    </option>
                    <option
                                        value="none">
                                        None
                    </option>
</select>,
                  "className": "book-shelf-changer",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <option
                        value="none">
                        Move to...
</option>,
                      <option
                        value="currentlyReading">
                        Currently Reading
</option>,
                      <option
                        value="wantToRead">
                        Want to Read
</option>,
                      <option
                        value="read">
                        Read
</option>,
                      <option
                        value="none">
                        None
</option>,
                    ],
                    "onChange": [Function],
                    "value": "read",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Move to...",
                        "value": "none",
                      },
                      "ref": null,
                      "rendered": "Move to...",
                      "type": "option",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Currently Reading",
                        "value": "currentlyReading",
                      },
                      "ref": null,
                      "rendered": "Currently Reading",
                      "type": "option",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Want to Read",
                        "value": "wantToRead",
                      },
                      "ref": null,
                      "rendered": "Want to Read",
                      "type": "option",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Read",
                        "value": "read",
                      },
                      "ref": null,
                      "rendered": "Read",
                      "type": "option",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "None",
                        "value": "none",
                      },
                      "ref": null,
                      "rendered": "None",
                      "type": "option",
                    },
                  ],
                  "type": "select",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "The Linux Command Line",
              "className": "book-title",
            },
            "ref": null,
            "rendered": "The Linux Command Line",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": "William E. Shotts, Jr.",
              "className": "book-authors",
            },
            "ref": null,
            "rendered": "William E. Shotts, Jr.",
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
